ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java

```diff
@@ -93,14 +118,19 @@ public long writeTo(WritableByteChannel channel) throws IOException {
       if (bufferToSend == null) {
         // this is the first time this method was called, prepare the buffer to send the header and other metadata
         // (everything except the blob content).
-        bufferToSend = ByteBuffer.allocate(sizeExcludingBlobSize());
+        bufferToSend = ByteBuffer.allocate(sizeExcludingBlobAndCrcSize());
         writeHeader();
+        int crcStart = bufferToSend.position();
         bufferToSend.put(blobId.toBytes());
         BlobPropertiesSerDe.putBlobPropertiesToBuffer(bufferToSend, properties);
         bufferToSend.putInt(usermetadata.capacity());
         bufferToSend.put(usermetadata);
         bufferToSend.putShort((short) blobType.ordinal());
         bufferToSend.putLong(blobSize);
+        crc.update(bufferToSend.array(), bufferToSend.arrayOffset() + crcStart, bufferToSend.position() - crcStart);
```

Comment:
trying to reason about what happens if we were to change the crc generation (add/remove fields). Assuming the replication hack is removed, during deployment, this would lead to an increase in the number of `Blob_Already_Exists` errors until all the frontends get the new code? Any other effects?